dapps/marketplace/src/hoc/withPartnerCampaignConfig.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from 'react'
+
+import { PARNTER_REFERRAL_CONFIG_URL } from 'constants/config'
+
+let cachedCampaigns
+
+function withPartnerCampaignConfig(WrappedComponent) {
+  const WithPartnerCampaignConfig = props => {
+    const [partnerConfig, setPartnerConfig] = useState(cachedCampaigns)
+
+    useEffect(() => {
+      let timeout
+      if (cachedCampaigns) {
+        return
+      } else {
+        setTimeout(async () => {
+          const resp = await fetch(PARNTER_REFERRAL_CONFIG_URL)
+          if (resp !== 200) {
+            console.error('Failed to fetch campaign config')
+            return
+          }
```

Comment:
@mikeshultz Just a thought. We seem to already have the code to fetch the referral campaigns in mobile and growth packages. Instead of having the similar code across packages, perhaps we can extract this into a separate GraphQL query and use that in all 3 (mobile, growth, DApp).

Reply:
Similar fetch in mobile: https://github.com/OriginProtocol/origin/blob/master/infra/growth/src/resources/rules/partnerReferralRule.js#L82

And in growth: https://github.com/OriginProtocol/origin/blob/master/mobile/src/screens/partnerWelcome.js#L62

Reply:
Adding a whole graphql resolver and fetch just to pull config JSON from originprotocol.com really irks me for some reason.  That said, I'm having some issues with `useEffect()` working properly so I'm amenable...

Reply:
Yeah, using fetch inside useEffect doesn't always go well (I learnt it the hard way with Split). 

For the DApp alone, you could move that to a GraphQL resolver (local schema, not the remote one) and handle the memory cache (the 60-minutes cache you do) in the resolver or set `fetchPolicy` to `cache-only` (since it is not bound to change often). In the HOC, you could use `useQuery` hook. Just my opinion though.

Reply:
Or you could just create a React Context for fetching that. Both should work fine.

Reply:
eh, I got the hook to work okay, I'm not going to refactor everything to integrate into graphql for the sake of speed.  Feel free to do that after if you feel strongly about it.